docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,438 @@
+nodes (DA nodes)` whose sole purpose is to interact with the `data availability
+network layer (DA layer)` such that the node contains all functionality
+necessary to post and retrieve messages from the DA layer.
+
+This means that DA nodes will be able to query for / modify celestia state such
+that the DA nodes are able to pay for posting their messages on the network.
+The state-related API will be documented below in detail.
+
+Furthermore, interaction between the celestia consensus network and the
+celestia data availability network will be the responsibility of the **bridge**
+node type. However, that interaction will not be exposed on a public level
+(meaning a **bridge** node will not expose the same API as the
+celestia-core node to which it is connected). A **bridge** node, for all intents
+and purposes, will provide the same API as that of a **full** node (with a
+stubbed-out DAS module as bridge nodes do not perform sampling).
+
+### Details
+
+#### Services Deprecation
+
+The initial step is to deprecate services in favor of modules. Ex.
+`HeaderService` -> `HeaderModule`.
+
+- We're organically moving towards the direction of modularized libraries. That
+is, our `share`, `header` and `state` packages are getting shaped as independent
+modules which now lack their own API definition.
+- Consistency. Semantically, modules are closer to what Celestia's overarching
+project goals and slogans.
+- Disassociate centralization. Services have always been associated with
+centralized infrastructure with older monolithic services and newer distributed
+microservice architectures.
+
+#### Modules Definition
+
+##### Header
+
+```go
+type HeaderModule interface {
+// Head returns the node's local head (tip of the chain of the header store).
+Head(ctx context.Context) (*header.ExtendedHeader, error)
+// Get returns the header of the given hash from the node's header store.
+Get(ctx context.Context, hash tmbytes.HexBytes) (*header.ExtendedHeader, error)
+// GetByHeight returns the header of the given height from the node's header store.
+// If the header of the given height is not yet available, the request will hang
+// until it becomes available in the node's header store.
+GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error)
+// GetRangeByHeight returns the given range [from:to) of ExtendedHeaders.
+GetRangeByHeight(ctx context.Context, from, to uint64) ([]*ExtendedHeader, error)
+// Subscribe creates long-living Subscription for validated ExtendedHeaders.
+// Multiple Subscriptions can be created.
+Subscribe() (Subscription, error)
+// SyncState returns the current state of the header Syncer. 
+SyncState() sync.State
+// SyncWait blocks until the header Syncer is synced to network head. 
+SyncWait(ctx context.Context) error
+// SyncHead provides the Syncer's view of the current network head.
+SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
```

Comment:
This is not correctly/consistently formatted. Compare `SharesModule` code-block below for instance.